flang/include/flang/Optimizer/Analysis/AliasAnalysis.h

```diff
@@ -153,17 +153,16 @@ struct AliasAnalysis {
     /// Return true, if Target or Pointer attribute is set.
     bool isTargetOrPointer() const;
 
-    /// Return true, if the memory source's `valueType` is a reference type
-    /// to an object of derived type that contains a component with POINTER
-    /// attribute.
-    bool isRecordWithPointerComponent() const;
-
     bool isDummyArgument() const;
     bool isData() const;
     bool isBoxData() const;
 
     mlir::Type getType() const;
 
+    /// Return true, if `ty` is a reference type to an object of derived type
+    /// that contains a component with POINTER attribute.
+    static bool isRecordWithPointerComponent(mlir::Type ty);
```

Comment:
nit: it may be better to make it a private static member of the `AliasAnalysis` class.

Reply:
Same for the `isPointerReference` member function that follows it?

Reply:
Done. Thanks for the review.